sim.c

```diff
@@ -506,25 +506,19 @@ END_OPERATOR
 
 BEGIN_OPERATOR(increment)
   LOWERCASE_REQUIRES_BANG;
+  PORT(0, -1, IN);
   PORT(0, 1, IN);
-  PORT(0, 2, IN);
   PORT(1, 0, IN | OUT);
-  Usz a = index_of(PEEK(0, 1));
-  Usz b = index_of(PEEK(0, 2));
+  Glyph g = PEEK(0, -1);
+  Usz rate = 1;
+  if (g != '.' && g != '*')
```

Comment:
I don't know how this new logic for `generator` is supposed to work. Is there some reason you have to branch here instead of just adding 1 to the result of `index_of(g)`?

Reply:
it is for `increment`, as i stated in the commit now lowercase operators can be banged from any direction with the default value, usually it is 0 but in case of `I` default is 1 so i had to add that extra branch for defaulting to 1 when a bang is received in the left port, otherwise it would have been ineffective.

Reply:
so when the input of the right port is not `*` or `.`,  0 is still a valid value for the left input and it will stop `I` from incrementing, but not as default value.